crates/dbsp/src/trace/spine_async/mod.rs

```diff
@@ -534,8 +557,16 @@ where
 
         for (level, m) in mergers.iter_mut().enumerate() {
             if let Some(merger) = m.as_mut() {
-                let mut fuel = 10_000;
+                let starting_fuel = if level == 0 {
```

Comment:
It would be helpful to explain the intuition behind this in a comment. It trust that this helps the performance, I'm just not sure why.

Reply:
I've got no idea, I was experimenting and it helped and @lalithsuresh asked me to merge it immediately.